parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -462,8 +480,17 @@ installKubeletKubectlAndKubeProxy() {
     # if the custom url is not specified and the required kubectl/kubelet-version via private url is not installed, install using the default url/package
     if [[ ! -f "/usr/local/bin/kubectl-${KUBERNETES_VERSION}" ]] || [[ ! -f "/usr/local/bin/kubelet-${KUBERNETES_VERSION}" ]]; then
         if [[ "$install_default_if_missing" == true ]]; then
+            if [[ ! -z ${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER} ]]; then
```

Comment:
per ALlison, we want to prefer using the OutboundNone/OutboundBlock .    This will tell us we need to use ORAS, and when there is a BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER define, we will overwrite the stored REGISTRY url with the one provided by the user.

Reply:
https://github.com/Azure/AgentBaker/pull/4867#discussion_r1735415421

According to this comment that they resolved, they want to go with BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER for the flag on which to use oras. And if that is the case, we should discuss if BLOCK_OUTBOUND_NETWORK should be removed.

Reply:
From my side BLOCK_OUTBOUND_NETWORK is still used for skipping outbound connection check, so that we do not need to remove?

Reply:
To remove confusion,
From aks api, we split network isolated cluster into two part, one is outbound type, another is bootstrap profile.
The outbound type defines if we block the egress traffic.
The bootstrap profile defines the artifact source cache or direct (if it is from acr cache).

The bootstrap profile artifact source cache is prerequisite of outbound type none/blocked.
However, we still allow customer to individually set bootstrap profile artifact source as Cache when outbound type is not none/blocked. (defined in PRD).
In this case, we allow customer egress traffic, it is not one complete network isolated cluster, but we may need to switch the image/binary source to acr cache rule, which is. the reason that we use BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER instead of BLOCK_OUTBOUND_NETWORK as the condition here.